### ISSUE TITLE
Issue updating Production Deployment

### ISSUE THREAD

**Issue by mdsimpson42** [colouring-cities/colouring-core#1129]
I've logged onto the production VM and tried running the last command that was run previously:

`cd ~/colouring-london/ && git fetch && git checkout origin/master && cd ~/scripts/ && ./build.sh && ./predeploy.sh && ./deploy.sh`

However, the app is not updating. I've deleted the ~/predeploy/app directory, but the new build version is not updating to show new features. What am I missing? I fixed the git origin, and it seems to have the updated code, but - for example - it isn't changing the name from "Colouring Cities" to "Colouring London", despite it having "London" specified in app/src/cc-config.json.

I need to be able to update the deployment on production and staging as I add new features, and we should really update the documentation for this for whoever takes over in future.

**Comment by polly64** [comment]
thanks Mike

**Comment by matkoniecz** [comment]
This command should work and it matches https://github.com/colouring-cities/colouring-london-config

No specific idea why config file is not being read in this case, or different is being read.

**Comment by matkoniecz** [comment]
I guess that adding logging to code that reads config file would be my next step, though maybe there is a smarter or more obvious solution.

But reporting which location it opens to read config would be helpful, I think - and would allow to check what is going on (is config file being read? Or is it being read and something else fails?).

**Comment by polly64** [comment]
thanks @matkoniecz @mdsimpson42  just to say staging saying database error when i add data but realise you'll be off soon.

**Comment by matkoniecz** [comment]
@polly64 On which action? Are you on staging or on colouring.london?

**Comment by polly64** [comment]
@just tried staging- clicked on building, selected option from dropdown and says database error and won't let me enter

**Comment by polly64** [comment]
@matkoniecz  survival (Age and History)

**Comment by matkoniecz** [comment]
That is because on staging database was not migrated and 

> column "survival_status" of relation "buildings" does not exist

**Comment by matkoniecz** [comment]
@polly64 I have run migration on staging

**Comment by matkoniecz** [comment]
(on production I would be more worried and would confirm first but staging should be fine for risky experiments)

Now it seems to work.

**Comment by polly64** [comment]
@matkoniecz thanks v much i can see it now but it won't save the edit , will it save it for you?

**Comment by matkoniecz** [comment]
@polly64 

Hmm. It works for me. 

https://cl-staging.uksouth.cloudapp.azure.com/edit/age/1207469

![screen01](https://user-images.githubusercontent.com/899988/230442783-79da308e-d5ae-4c9a-805a-8b38ad572968.png)

> btw if i work on staging can my edits be copied and pasted onto main site later?

not easily, especially when edit histor on objects should be kept

**Comment by matkoniecz** [comment]
@mdsimpson42 On main site error "Field does not exist" is appearing.

**Comment by polly64** [comment]
@weird i realise it is loading the entry but building remains blank until i zoom out and then it colours=

**Comment by matkoniecz** [comment]
Also, next to Euston railway station

![Peek 2023-04-06 18-47](https://user-images.githubusercontent.com/899988/230443791-c62df860-26a9-4ab8-b159-e9106b0198d0.gif)

**Comment by polly64** [comment]
![survival 1b](https://user-images.githubusercontent.com/42236514/230444350-94ea70a8-2687-4221-a207-b932a01116d3.JPG)
![survival 1a](https://user-images.githubusercontent.com/42236514/230444361-a2c2df15-762d-41e6-9e8c-2ae4cfc5d586.JPG)

**Comment by matkoniecz** [comment]
I see, maybe new layer was supposed to be on top of historic map but has the same priority so it is randomly partially above and partially below?

**Comment by polly64** [comment]
ok thanks for your help!  will chat to Mike and try to sort this next week

**Comment by matkoniecz** [comment]
Maybe even close this issue and create new ones for all things mentioned here and not solved, including original issue.

> site not building correctly on production

Is not reading config file the only known problem in the end? If that is present - have you tried logging point when it is supposed to read config file?

**Comment by mdsimpson42** [comment]
> Is not reading config file the only known problem in the end? If that is present - have you tried logging point when it is supposed to read config file?

No, its not just the config file. It's like the code is out of date, but I have downloaded the latest version. One of the things I checked is that it says "London" not "Cities" in the config file, so the deployed version on colouring.london should says "Colouring Cities", but it doesn't. It also doesn't have the ability to select "survival status" on the key in the Age/History category, which it should have. 

I deleted the temporary files, which get copied by deploy.sh, so it is being rebuilt from scratch, but it's like it's being built from old code. There were no obvious warnings or error messages...

I'll have another look again tomorrow, and go step by step through all the code in those scripts.
